File: 01_projects/code/java-playwright/cbaps-dawms/robust/src/main/java/com/playwright/cbaps/pages/CBAPSDashboardPage.java
package com.playwright.cbaps.pages;
import com.microsoft.playwright.*;
import com.playwright.cbaps.library.EnhancedPlaywrightManager;
import org.slf4j.*;

public class CBAPSDashboardPage {
    private static final Logger log = LoggerFactory.getLogger(CBAPSDashboardPage.class);
    private Page page;
    private EnhancedPlaywrightManager pwm;
    private Locator createRequisitionBtn;
    
    public CBAPSDashboardPage(Page page, EnhancedPlaywrightManager pwm) {
        this.page = page; this.pwm = pwm;
        this.createRequisitionBtn = page.locator("button:has-text('Create Requisition')");
        pwm.waitUntilElementVisible("button:has-text('Create Requisition')");
    }
    
    public RequisitionPage goToCreateRequisition() {
        pwm.clickElement(createRequisitionBtn);
        return new RequisitionPage(page, pwm);
    }
}

File: 01_projects/code/python-playwright/cbaps-dawms/robust/cbaps/pages/cbaps_dashboard_page.py
from playwright.sync_api import Page
from loguru import logger

class CBAPSDashboardPage:
    def __init__(self, page: Page, pw):
        self.page = page
        self.pw = pw
        self.create_req_btn = page.locator("button:has-text('Create Requisition')")
        self.pw.wait_for_selector("button:has-text('Create Requisition')")
        logger.info("✅ CBAPS Dashboard loaded")
    
    def go_to_create_requisition(self):
        from .requisition_page import RequisitionPage
        self.pw.click_element(self.create_req_btn)
        logger.info("➡️ Create Requisition")
        return RequisitionPage(self.page, self.pw)
